Libraries/TVector/TVector.h

```diff
@@ -0,0 +1,151 @@
+#ifndef TVector_h
+#define TVector_h
+#include "Tile.h"
+
+template <class T>
+class TVector {
+  private:
+    uint8_t iSize;
+    T *vector;
+
+  public:
+    TVector();//constructor with 1 element
+    TVector(const TVector &V);//constructor that copies another vector
+    ~TVector();
+
+    uint8_t getSize();
+
+    void pushTop(T t);
+    void pushFront(T t);
+    void popTop();
+    void popFront();
+    void set(T t, uint8_t index);
+    void erase(uint8_t index);
+
+    T& operator[](uint8_t index) const;
+    void operator=(const TVector &V);
+};
+
+template <class T>
+TVector<T>::TVector() {
+  iSize = 0;
+  vector = NULL;
+}
+
+template <class T>
+TVector<T>::TVector(const TVector &V) {
+  vector = new T[V.iSize];
+  
+  for(int i = 0 ; i < V.iSize ; i++) {
```

Comment:
++i, not i++.
https://stackoverflow.com/questions/24853/c-what-is-the-difference-between-i-and-i

Even though most modern compilers optimize it, try and always follow the standard.